### PR TITLE
Adding Tournament cpp & WinRT APIs

### PR DESCRIPTION
**Pull request by adityatMS**
- Adding get_tournaments & get_tournament_details APIs
- Added UnitTests for the tournament APIs

### CODE REVIEW

File: Source/Services/Tournaments/WinRT/TournamentService_WinRT.cpp [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+//        STRING_T_FROM_PLATFORM_STRING(teamId)
+//        )
+//    .then([](xbox_live_result<team_details> cppResult)
+//    {
+//        THROW_IF_ERR(cppResult);
+//        return ref new TeamDetails(cppResult.payload());
+//    });
+//
+//}
+
```
Comment (jasonsandlin) [inline Source/Services/Tournaments/WinRT/TournamentService_WinRT.cpp]: remove?

File: Source/Services/Tournaments/WinRT/TournamentService_WinRT.h [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    /// <remarks>
+    /// Returns a concurrency::task<T> object that represents the state of the asynchronous operation.
+    /// This method calls GET /tournaments/{organizer}/{id}/teams/{teamId}
+    /// </remarks>
+    /*Windows::Foundation::IAsyncOperation<TeamDetails^>^ GetTeamDetails(
+        _In_ Platform::String^ organizerId,
+        _In_ Platform::String^ tournamentId,
+        _In_ Platform::String^ teamId
+        );*/
+
```
Comment (jasonsandlin) [inline Source/Services/Tournaments/WinRT/TournamentService_WinRT.h]: remove?

File: Tests/UnitTests/Tests/Services/TournamentsTests.cpp [hunk 3]
```diff
@@ -1,210 +1,154 @@
...
+//*********************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
+// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
+// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
+// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
+//
+//*********************************************************
 #include "pch.h"
```
Comment (jasonsandlin) [inline Tests/UnitTests/Tests/Services/TournamentsTests.cpp]: revert header

File: Tests/UnitTests/Tests/Services/TestResponses/Tournaments.json [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+                "hasPrize": false,
+                "paused": false,
+                "visibility": "Public",
+                "maxRegistrationCount": 2,
+                "minRegistrationCount": 2,
+                "maxWaitlistCount": 0,
+                "minTeamSize": 1,
+                "maxTeamSize": 1,
+                "schedule": {
+                    "registrationStart": "2017-02-17T20:06:10.1949991Z",
```
Comment (edjamesmsft) [inline Tests/UnitTests/Tests/Services/TestResponses/Tournaments.json]: >"registrationStart": "2017-02-17T20:06:10.1949991Z", [](start = 20, length = 52)

format of this timestamp and the above are not the same - does this matter?

Comment (adityatMS) [inline Tests/UnitTests/Tests/Services/TestResponses/Tournaments.json]: Good catch. Thanks.

---
In reply to: [106509728](https://github.com/Microsoft/xbox-live-api/pull/100#discussion_r106509728) [](ancestors = 106509728)

**Comment by edjamesmsft** [comment]
:shipit:

**Pull request merged**